File: May-18/14-05-18/python/01-Decode.py
for _ in range(int(input())):
    data = input()
    n = len(data)
    ch = [0]*n
    cnt = n - 1
    
    for i in range(n-1, -1, -2):
        ch[cnt] = data[i]
        cnt -= 1
    extra = 0
    if(n % 2 != 0):
        extra = 1
    
    for i in range(0 + extra, n, 2):
        ch[cnt] = data[i]
        cnt -= 1
    
    for i in ch:
        print(i, end = '')
    print()

File: May-18/14-05-18/c++/01-Decode.cpp
#include<bits/stdc++.h>

using namespace std;

int main()
{
    int t;
    cin >> t;
    while(t--)
    {
        string str;
        cin >> str;
        int n = str.length();
        
        string ch[n];
        int cnt = n - 1;
        
        for(int i = n - 1; i >= 0; i -= 2)
        {
            ch[cnt] = str[i];
            cnt--;
        }
        
        
        int extra = 0;
        if(n % 2 != 0)
            extra += 1;
        
        for(int i = 0 + extra; i < n; i += 2)
            {
                ch[cnt] = str[i];
                cnt--;
            }
        
        for(int i = 0; i < n; i++)
        {
            cout << ch[i];
        }
        cout << endl;
        
    }
    
}
